### DISCUSSION TITLE
How to start contributing?

### DISCUSSION THREAD

**Discussion by lccambiaghi** [ProcursusTeam/Procursus#688] [Q&A] [answered]
Hi, I own a JB iPad Pro and I would like to install some tools that I use on my Mac on it. How do I begin? Do you have any material/resources I can read?

I (kinda) understand the concept of cross-compilation, where we target the iphoneos-arm architecture instead of whatever it is targeted for building on a Mac. We build a .deb and install it on the iOS device. 

How do I get from the tool working on macOS to the tool working on iOS? In practice, how do I know what to write in the Makefile??

Thank you for this amazing project!!

EDIT:
I have done some more research, I have read the main Makefile and compiled and packaged my first tool from the repo. 
To answer my own question: I guess one way to come up with the build target could be to look at brew formulas? I would still love to read an 'introductory guide' or something, which would teach good practices and common strategies for cross-compiling to iOS

**Comment by m1stadev** [comment, marked as answer]
a good first place to look is [CONTRIBUTING.md](https://github.com/ProcursusTeam/Procursus/blob/as2/CONTRIBUTING.md), but probably is a bit outdated. i'd recommend looking there (more specifically, at [grep.mk.template](https://github.com/ProcursusTeam/Procursus/blob/as2/grep.mk.template)) just to get a gist of how the makefiles work, then looking at some current makefiles (e.g. [nano.mk](https://github.com/ProcursusTeam/Procursus/blob/as2/nano.mk)) as a starting guide to adding whatever package you're interested in adding. if you have any other questions, feel free to join the support Discord [here](https://diatr.us/discord), where we actively discuss Procursus work being done.

**Reply by CRKatri** [reply]
FYI, `grep.mk.template` is very outdated, to start adding something I recommend using `./build_tools/new_package.sh` to get started as well, instead of copying some other makefile.

**Comment by TheRealKeto** [comment]
When it comes to adding a package to Procursus, specific functions, which can be found in the main Makefile, are used in order to perform a specific action. These are meant to be documented, but that has been in slow development for a while now, since Procursus constantly changes as others contribute and add new packages/new features.

There's no specific "do this here" step primarily because each package is different. There are examples that can get you started, but ultimately, you have to know what package you're adding, how it works, and how it must be compiled in order for you to write the specific action that Make (the tool used to package Procursus packages) must do in order to compile a given tool. 

As suggested by adam, you can look at the README of the project, as well as the contribution guidelines to get an idea as to how your package must be done. On top of that, you should join the Discord (given by adam) in order to ask further questions.
